src/cudamatrix/cu-matrix-test.cc

```diff
@@ -478,6 +600,46 @@ static void UnitTestCuMatrixSumColumnRanges() {
 }
 
 
+template<typename Real>
+static void UnitTestCuMatrixAddRowRanges() {
+  for (MatrixIndexT p = 0; p < 10; p++) {
+    MatrixIndexT num_rows1 = 10 + Rand() % 10,
+        num_rows2 = 10 + Rand() % 10,
+        num_cols = 10 + Rand() % 10;
+    Matrix<Real> src(num_rows1, num_cols); src.SetRandn();
+    Matrix<Real> dst(num_rows2, num_cols); dst.SetRandn();
+
+    // Computes the indexes.
+    std::vector<Int32Pair> indexes(num_cols);
+    for (MatrixIndexT i = 0; i < num_cols; i++) {
+      indexes[i].first = Rand() % num_rows1;
+      int32 headroom = num_rows1 - indexes[i].first,
```

Comment:
Probably MatrixIndexT. The type of `indexes` is also, strictly speaking, a pair of `MatrixIndexT`s.